xarray/testing/assertions.py

```diff
@@ -104,7 +116,7 @@ def assert_equal(a: DataTree, b: DataTree, from_root: bool = True): ...
 
 
 @ensure_warnings
-def assert_equal(a, b, from_root=True):
+def assert_equal(a, b, from_root=True, check_dims="strict"):
```

Comment:
Let's call this kwarg `check_dim_order` to clarify that the only thing it affects is whether or not the order of the dimensions is checked (not e.g. whether or not dimensions are checked in general).
```suggestion
def assert_equal(a, b, from_root=True, check_dim_order="strict"):
```

Reply:
Can we go with `check_dim_order: bool` so only `True` and `False` as input instead? `False` implies a different order is OK. Perhaps `strict_dim_order` is an alternate name.